compiler/rustc_codegen_llvm/src/llvm/ffi.rs

```diff
@@ -645,16 +632,6 @@ pub enum ThreadLocalMode {
     LocalExec,
 }
 
-/// LLVMRustTailCallKind
-#[derive(Copy, Clone)]
-#[repr(C)]
-pub enum TailCallKind {
-    None,
-    Tail,
-    MustTail,
-    NoTail,
-}
```

Comment:
Remark: This removal was justified in https://github.com/rust-lang/rust/pull/135502#issuecomment-2593582096, so good to remove.